sdk/servicebus/Azure.Messaging.ServiceBus/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,305 @@
+
+Further reading:
+
+Debug ThreadPool Starvation  
+Diagnosing .NET Core ThreadPool Starvation with PerfView (Why my service is not saturating all cores or seems to stall)  
+Diagnosing ThreadPool Exhaustion Issues in .NET Core Apps (video)
+
+### Session processor takes too long to switch sessions
+
+This can be configured using the [SessionIdleTimeout][SessionIdleTimeout], which tells the processor how long to wait to receive a message from a session, before giving up and moving to another one. This is useful if you have many sparsely populated sessions, where each session may only have a few messages. If you expect that each session will have many messages that trickle in, setting this too low can be counter productive, as it will result in unnecessary closing of the session.
+
+### Processor stops immediately
+
+This typically comes up for demo or testing scenarios. Calling StartProcessingAsync returns immediately after the processor has started. Calling this method will not keep your app alive for as long as the processor is running, so you'll need some other mechanism to keep the app alive until you wish to shut it down. For demos or testing, it may be sufficient to just add a Console.ReadKey() call after you start the processor. For production scenarios, you will likely want to use some sort of framework integration like [BackgroundService][BackgroundService] to provide convenient application lifecycle hooks that can be used for starting and disposing the processor.
+
+## Troubleshoot transactions
+
+For general information about transactions in Service Bus, see the [Overview of Service Bus transaction processing][Transactions].
+
+### Supported operations
+
+Not all operations are supported when using transactions. To see the list of supported transactions, see [Operations within a transaction scope][TransactionOperations].
+
+### Timeout
+
+A transaction times out after 2 minutes. The transaction timer starts when the first operation in the transaction starts.
```

Comment:
Is there service documentation we can point to for this rather than keeping the value here?